drivers/i2c/i2c_litex.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include "i2c_bitbang.h"
+
+#define SCL_BIT_POS                0
+#define SDA_DIR_BIT_POS            1
+#define SDA_BIT_W_POS              2
+#define SDA_BIT_R_POS              0
+
+#define SDA_DIR_OUTPUT             1
+#define SDA_DIR_INPUT              0
+
+#define HIGH_STATE_ON_I2C_LINES    0x7
+
+struct i2c_litex_cfg {
+	volatile u32_t *w_reg;
+	volatile u32_t *r_reg;
```

Comment:
This register is only read, and it seems to be not supposed to be written (at least through this pointer), so the pointer should be `volatile const u32_t *`.

Reply:
Changed as suggested